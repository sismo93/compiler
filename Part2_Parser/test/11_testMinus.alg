co gives us the right value even with several minus
begin
  a := ----------5
end
