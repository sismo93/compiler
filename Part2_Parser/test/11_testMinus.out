Rules used : 1 3 4 7 13 14 15 16 16 16 16 16 16 16 16 16 16 18 23 21 6 
 Here are all the grammar used with more detail : 

<Program> -> begin <Code> end

<Code> -> <InstList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <Assign>

<Assign> -> [VarName] := <ExprArith>

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> MINUS <Atom>

<Atom> -> MINUS <Atom>

<Atom> -> MINUS <Atom>

<Atom> -> MINUS <Atom>

<Atom> -> MINUS <Atom>

<Atom> -> MINUS <Atom>

<Atom> -> MINUS <Atom>

<Atom> -> MINUS <Atom>

<Atom> -> MINUS <Atom>

<Atom> -> MINUS <Atom>

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<InstListPrim> -> EPSILON


